### PR TITLE
wildcard and pagination on devicenames

### PR DESCRIPTION
**Pull request by meguiraun**
* faster retrieval of devices
* wilcard
* pagination

e.g.
`...../devices?range=500-554`
`...../devices?range=0-10&wildcard=tg_test`

Missing:
* improve the regex for the domain/family/member
* on pagination enhance the response according to the standard (code 206 and so on)

### CODE REVIEW

File: api/rc3.py [hunk 1]
```diff
@@ -89,15 +89,18 @@
...
-	for d in domains:
-		families = db.get_device_family("%s/*" % d)
-		for f in families:
-			members = db.get_device_member("%s/%s/*" % (d, f))
-			for m in members:
-				devs.append('/'.join((d, f, m)))
+
+	dev_range = rq.args.pop("range", None)
+	filters = rq.args
+	db_proxy = await getDeviceProxy('sys/database/2')
```
Comment (hardion) [inline api/rc3.py]: 'sys/database/2' is not always the database
you can use db (PyTango.Database()) which give you a proxy on the default database. You may not profit of the async method but the previous code did not use it.

File: utils.py [hunk 2]
```diff
@@ -27,3 +29,23 @@ def __str__(self):
 
 	def url(self):
 		return "/".join(self._as_str_tuple())
+
+
+def device_filtering(devices, filters=None, range=None):
```
Comment (hardion) [inline utils.py]: missing doc

File: api/rc3.py (hunk 1 above)
Comment (meguiraun) [reply api/rc3.py]: `db` does not give a method for getting the devices list. is there any way of guessing the db device name from the `db`?

Comment (meguiraun) [reply api/rc3.py]: `db.get_name()`

File: api/rc3.py [hunk 3]
```diff
@@ -98,7 +99,7 @@
 
 	dev_range = rq.args.pop("range", None)
 	filters = rq.args
-	db_proxy = await getDeviceProxy('sys/database/2')
+	db_proxy = await getDeviceProxy(db.get_name())
```
Comment (hardion) [inline api/rc3.py]: This is perfect. You resolve both issue, a default database and still able to do async request.

**Comment by hardion** [comment]
Thanks for your contribution.

**Pull request merged**